docs/en/tutorials/getting-started.rst

```diff
@@ -233,33 +233,154 @@ entity definition:
          * @var string
          */
         protected $name;
+    }
 
-        public function getId()
-        {
-            return $this->id;
+When creating entity classes, all of the fields should be protected or private (not public).
+
+Adding the Entity behaviors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You have two options to define the accessors mutators that will
+update the state of your objects.
+
+**Getters and setters**
+
+The most popular method is to create getter and setter methods
+for each property (except ``$id``). The use of mutators allows
+Doctrine to hook into calls which manipulate the entities in ways
+that it could not if you just directly set the values with ``entity#field = foo;``.
+
+The id field has no setter since, generally speaking, your code
+should not set this value since it represents a database id value.
+(Note that Doctrine itself can still set the value using the
+Reflection API instead of a defined setter function)
+
+.. note::
+
+    This method, although very common, is **not** the best approach if
+    you want a perfect domain design. It can lead to unexpected behavior
+    when dealing with different steps of logic in between different
+    calls to ``EntityManager::flush()``, for example when you change
+    the state of an object via event listeners or lifecycle callbacks.
+
+**Mutators and DTOs**
+
+We recommend using a rich model design and rely on more complex
+mutators, ideally based on DTOs.
+This way, you should **not** use getters nor setters, and instead,
+implement methods that represent the **behavior** of your domain.
+
+For example, when having a ``User`` entity, we could foresee
+the following kind of optimization.
+
+Before, an anemic model with getters and setters:
```

Comment:
there are sections "Getting Started: Database First" and "Getting Started: Model First". I think this sections could cover "anemic vs rich model" cases.

Reply:
I think it's a.... rich idea :sunglasses: